flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/HBaseTableSource.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.addons.hbase;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.table.sources.BatchTableSource;
+import org.apache.flink.table.sources.ProjectableTableSource;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Creates a table source that helps to scan data from an hbase table
+ *
+ * Note : the colNames are specified along with a familyName and they are seperated by a ':'
+ * For eg, cf1:q1 - where cf1 is the familyName and q1 is the qualifier name
+ */
+public class HBaseTableSource implements BatchTableSource<Row>, ProjectableTableSource<Row> {
+
+	private Configuration conf;
+	private String tableName;
+	private byte[] rowKey;
+	private String[] colNames;
+	private TypeInformation<?>[] colTypes;
+
+	public HBaseTableSource(Configuration conf, String tableName, byte[] rowKey, String[] colNames,
+							TypeInformation<?>[] colTypes) {
+		this.conf = conf;
+		this.tableName = Preconditions.checkNotNull(tableName, "Table  name");
+		this.rowKey = Preconditions.checkNotNull(rowKey, "Rowkey");
```

Comment:
`new RowTypeInfo(
	new TypeInformation[]{
		new RowTypeInfo(
			new TypeInformation[]{
				BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.INT_TYPE_INFO},
			new String[]{"name", "age"})
	},
	new String[]{"person"}
);`
I did the above change for the HBaseTableSource's return type and then made a simple query
`		tableEnv.registerTableSource("test", hbaseTable);
		Table result = tableEnv
			.sql("SELECT f1.q1, f1.q2, f1.q3 FROM test");`
It throws sql validation error and considers f1 as table name and says tableName 'f1' not found.